diff --git a/bsp/at32/libraries/AT32F435_437_Firmware_Library/drivers/inc/at32f435_437_can.h b/bsp/at32/libraries/AT32F435_437_Firmware_Library/drivers/inc/at32f435_437_can.h
index 3dd9011386..f555a27f89 100644
--- a/bsp/at32/libraries/AT32F435_437_Firmware_Library/drivers/inc/at32f435_437_can.h
+++ b/bsp/at32/libraries/AT32F435_437_Firmware_Library/drivers/inc/at32f435_437_can.h
@@ -256,6 +256,14 @@ typedef enum
 /**
   * @brief  can receive fifo
   */
+#ifdef CAN_RX_FIFO0
+#undef CAN_RX_FIFO0
+#endif
+
+#ifdef CAN_RX_FIFO1
+#undef CAN_RX_FIFO1
+#endif
+
 typedef enum
 {
   CAN_RX_FIFO0                           = 0x00, /*!< can fifo 0 used to receive */
